#! /bin/sh /usr/share/dpatch/dpatch-run
## 30_bytesex.dpatch by Romain Francoise <rfrancoise@debian.org>
##
## All lines beginning with `## DP:' are a description of the patch.
## DP: Endianness fixes.

@DPATCH@
diff -urNad hping3-3.a2.ds1/bytesex.h /tmp/dpep.6fhOWO/hping3-3.a2.ds1/bytesex.h
--- hping3-3.a2.ds1/bytesex.h	2003-08-31 19:23:48.000000000 +0200
+++ /tmp/dpep.6fhOWO/hping3-3.a2.ds1/bytesex.h	2005-03-12 12:44:18.692860080 +0100
@@ -7,16 +7,11 @@
 #ifndef ARS_BYTESEX_H
 #define ARS_BYTESEX_H
 
-#if 	defined(__i386__) \
-	|| defined(__alpha__) \
-	|| (defined(__mips__) && (defined(MIPSEL) || defined (__MIPSEL__)))
+#include <endian.h>
+
+#if __BYTE_ORDER == __LITTLE_ENDIAN
 #define BYTE_ORDER_LITTLE_ENDIAN
-#elif 	defined(__mc68000__) \
-	|| defined (__sparc__) \
-	|| defined (__sparc) \
-	|| defined (__PPC__) \
-	|| defined (__BIG_ENDIAN__) \
-	|| (defined(__mips__) && (defined(MIPSEB) || defined (__MIPSEB__)))
+#elif __BYTE_ORDER == __BIG_ENDIAN
 #define BYTE_ORDER_BIG_ENDIAN
 #else
 # error can not find the byte order for this architecture, fix bytesex.h
